tools/cmake/toolchains/GCC_ARM.cmake

```diff
@@ -67,23 +67,23 @@ function(mbed_set_toolchain_options target)
 endfunction()
 
 # GCC ARM requires preprecessing linker script, execute generators to get definitions needed for
-# this step - linker options and compile definitions 
-function(mbed_generate_gcc_options_for_linker target definitions_file linker_options_file)
+# this step - compile definitions
+function(mbed_generate_gcc_options_for_linker target definitions_file)
     set(_compile_definitions 
         "$<TARGET_PROPERTY:${target},COMPILE_DEFINITIONS>"
     )
-
-    set(_linker_options
-        "$<TARGET_PROPERTY:${target},LINK_OPTIONS>"
+    # We filter -D to the linker script, as only simple ones are passed (value based)
```

Comment:
I would replace this whole section comment with something more descriptive of what is happening here, why we are doing it, and what we want to do in the future.
```
# Remove macro definitions that contain spaces as the lack of escape sequence and quotation marks
# in the macro when retrieved using generator expressions causes linker errors.
# This includes string macros, array macros, and macros with operations.
# TODO CMake: Add escape sequences and quotation marks where necessary instead of removing these macros. 
```

Reply:
fixed